### PR TITLE
[I18n] Single type core api update

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Support  Single type core api update

> Needs https://github.com/strapi/strapi/pull/9766 to be merged first

### CODE REVIEW

File: packages/strapi/lib/core-api/service/single-type.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+        return strapi.entityService.create({ data: sanitizedData, files }, { model: modelName });
+      } else {
+        return strapi.entityService.update(
+          {
+            params: {
+              id: entity.id,
+            },
+            data: sanitizedData,
+            files,
```
Comment (petersg83) [inline packages/strapi/lib/core-api/service/single-type.js:46]: should the files be sanitized too?

File: packages/strapi/lib/core-api/service/single-type.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        );
+      }
+    },
+
+    /**
+     * Deletes the single type content
+     *
+     * @return {Promise}
+     */
+    async delete() {
```
Comment (petersg83) [inline packages/strapi/lib/core-api/service/single-type.js:58]: i18n ST delete will be done in another PR

**Review by petersg83 — CHANGES_REQUESTED** [review]
Works well

File: packages/strapi/lib/core-api/service/single-type.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+     * @return {Promise}
+     */
+    async createOrUpdate(data, { files, query } = {}) {
+      const entity = await this.find(query);
+      const sanitizedData = sanitizeInput(data);
+
+      if (!entity) {
+        const count = await strapi.query(modelName).count();
+        if (count >= 1) {
+          throw strapi.errors.badRequest('singleType.alreadyExists');
```
[the remark this answers is not in the document]
Comment (alexandrebodin) [reply packages/strapi/lib/core-api/service/single-type.js:35]: Yeah this one is a bit tricky it would almost require a hook from the i18n plugin to handle this specific usecase... I feel like if the doc is good enough this will be ok good for the mvp

File: packages/strapi/lib/core-api/service/single-type.js (hunk 1 above)
Comment (alexandrebodin) [reply packages/strapi/lib/core-api/service/single-type.js:46]: I prefered not to change the current implementation here :)

**Review by petersg83 — APPROVED** [review]
LGTM :) It feels good to approach the end of the feature !
